tests/kafkatest/tests/core/security_rolling_upgrade_test.py

```diff
@@ -100,17 +103,58 @@ def roll_in_sasl_mechanism(self, security_protocol, new_sasl_mechanism):
         # Bounce again with ACLs for new mechanism
         self.set_authorizer_and_bounce(security_protocol, security_protocol)
 
+    @cluster(num_nodes=9)
+    @matrix(broker_protocol=[SecurityConfig.SSL, SecurityConfig.SASL_PLAINTEXT, SecurityConfig.SASL_SSL],
+            client_protocol=[SecurityConfig.SSL, SecurityConfig.SASL_PLAINTEXT, SecurityConfig.SASL_SSL])
+    def test_enable_separate_interbroker_listener(self, client_protocol, broker_protocol):
+        """
+        Start with a cluster that has a single {{client_protocol}} listener.
+        Start producer and consumer on the {{client_protocol}} listener.
+        Open a SECURED dedicated interbroker port via rolling upgrade.
+        Ensure we can produce and consume via {{client_protocol}} port throughout.
+        """
+        self.kafka.security_protocol = client_protocol
+        self.kafka.setup_interbroker_listener(client_protocol, use_separate_listener=False)
+        self.kafka.start()
+
+        # create plaintext producer and consumer
+        self.create_producer_and_consumer()
+
+        # run produce/consume/validate loop while enabling a separate interbroker listener via rolling restart
+        self.run_produce_consume_validate(self.setup_interbroker_listener, broker_protocol, True)
+
+    @cluster(num_nodes=9)
+    @matrix(broker_protocol=[SecurityConfig.SSL, SecurityConfig.SASL_PLAINTEXT, SecurityConfig.SASL_SSL],
+            client_protocol=[SecurityConfig.SSL, SecurityConfig.SASL_PLAINTEXT, SecurityConfig.SASL_SSL])
```

Comment:
Again, not sure why we have all thse combinations, and why the test is with upgrade tests.

Reply:
Agreed, will move it out.